### ISSUE TITLE
Ability to play Liked Songs starting from selected track 

### ISSUE THREAD

**Issue by jhouse1989** [thlucas1/spotifyplus_card#35]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/thlucas1/spotifyplus_card/issues?q=is%3Aissue+label%3A%22Feature+Request%22+).

### Is your feature request related to a problem? Please describe.

I can use the “track favorites” option but when I do it will only play the specific song I press on and not continue through the list after. 

### Describe the solution you'd like

Ideally it would work like Spotify app where you click on where you want to start and then it will automatically play the next song in the list. 

### Describe alternatives you've considered

Copying all songs from “liked” to an actual playlist. 

### Additional context

![IMG_6549](https://github.com/user-attachments/assets/aca880ac-e9ac-47a2-ba99-7019aa33b12d)
![IMG_6550](https://github.com/user-attachments/assets/fd293546-eee1-4e8e-b9d1-af54caa153d5)

I will click on one song from “track favorites” the equivalent of liked song on Spotify. Then in the player queue it will only include that song. Not the subsequent songs.

**Comment by thlucas1** [comment]
@jhouse1989 
Thanks for the suggestion.

Just released this - give it a try and let me know.

[ 1.0.25 ] - 2024/12/23

  * Ability to play track favorites starting from selected favorite track, which will now automatically add the following tracks (up to 50) to the player queue.  Prior to this, only the selected track would play and then play would stop.  Note that the 50 track limitation is a Spotify Web API limit.

**Comment by jhouse1989** [comment]
Thanks, I just tried it out. Unfortunately I get the following error: “ Failed to perform the action spotifyplus/player_media_play_tracks. Validation error: Device not found”.  When playing anything besides things in the track Favorite menu it still works as intended.

**Comment by thlucas1** [comment]
You have to select a device first if one is not already active

**Comment by jhouse1989** [comment]
I do have my echo show active and playing music. Unfortunately the error still comes up. 
![IMG_6565](https://github.com/user-attachments/assets/369c65fa-db5c-4b44-83c2-9a2f969af605)

**Comment by thlucas1** [comment]
Do you see the echo in the Spotify Card Devices section?  And is it the active device, as indicated by a little bar graph icon to the right of it.  

Here's an example of my device list, with the "Bose-ST10-1" being the active player.
![image](https://github.com/user-attachments/assets/2af69837-4872-4749-a4c5-147cf4ed0e86)

I'm not sure if the Amazon Echo devices show as Spotify Connect devices or not ... they should, if they are true Spotify Connect enabled devices.

**Comment by jhouse1989** [comment]
Yes, it shows in the spotify connect devices. and shows as playing.
<img width="632" alt="Screenshot 2024-12-23 at 1 23 27 PM" src="https://github.com/user-attachments/assets/a3b11600-2ff7-4172-87ee-0b0b70b6d723" />

**Comment by thlucas1** [comment]
Can you verify the SpotifyPlus Integration (not SpotifyPlus Card) version that you have installed?
You might try upgrading to v1.0.75 if you're not already there.

**Comment by jhouse1989** [comment]
I am running 1.0.75. It looks like the "Default Spotify Connect Player Device ID when none are active." needs to be used in order for this to work. When I selected a device for this it started working. Strange that only this button required that.

**Comment by thlucas1** [comment]
Ya, that is odd.  The "Default Spotify Connect Player Device ID when none are active." is forcing the selection of the device, because it detected that no device was active.  It should have detected that in the card.  

I will look into it a little more when I get time, but it's hard to determine 100% since I don't have an Echo device to test with - it could be doing something behind the scenes that I'm not seeing.

**Comment by thlucas1** [comment]
@jhouse1989 
I re-vamped some of the Spotify Connect Zeroconf processing with the latest v1.0.86 release of the underlying SpotifyPlus integration.  It also adds Zeroconf support for Google Chromecast devices.  I also updated the SpotifyPlus Card to v1.0.33 for these changes.

If you're still having issues with the Echo Dot, this may fix it.

Please give it a try when you get a chance, and let me know if it fixes it or not?

Thanks!

**Comment by thlucas1** [comment]
@jhouse1989 
FYI - just released a new version of the [SpotifyPlus integration](https://github.com/thlucas1/homeassistantcomponent_spotifyplus/wiki)

The big takeaway with this release the the ability to set the [Spotify Web Player Credentials in UI configuration options](https://github.com/thlucas1/homeassistantcomponent_spotifyplus/wiki/Device-Configuration-Options#spotify-web-player-cookie-credentials-sp_dc-sp_key), instead of in the Token Cache File.

Setting these credentials should allow you to do a simple `media_player.select_source` service call, which will re-activate the device if needed (no more calls to the SpotifyPlus zeroconf_X services).

Be sure to check out the [SpotifyPlus Card Dashboard](https://github.com/thlucas1/spotifyplus_card/wiki) as well.

Hope it helps!

**Comment by thlucas1** [comment]
@jhouse1989 
FYI - just released v1.0.109 of the [SpotifyPlus integration](https://github.com/thlucas1/homeassistantcomponent_spotifyplus/wiki), which fixes the `device not found` errors when playing on Amazon devices.  This problem was due to Amazon devices falling into "sleep" mode due to inactivity.

Give it a try when you get a chance, and let me know if this fixes your issue.
